src/ariel-os-debug/src/lib.rs

```diff
@@ -14,17 +18,36 @@ compile_error!(
     r#"feature "debug-console" enabled but no backend. Select feature "rtt-target" or feature "esp-println"."#
 );
 
-pub const EXIT_SUCCESS: Result<(), ()> = Ok(());
-pub const EXIT_FAILURE: Result<(), ()> = Err(());
-pub fn exit(code: Result<(), ()>) {
-    let code = match code {
-        EXIT_FAILURE => 1,
-        EXIT_SUCCESS => 0,
-    };
+/// Represents the exit code of a debug output.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum ExitCode {
+    #[doc(hidden)]
+    Success,
+    #[doc(hidden)]
+    Failure,
+}
+
+impl ExitCode {
+    /// The [`ExitCode`] for success.
+    pub const SUCCESS: Self = Self::Success;
+    /// The [`ExitCode`] for failure.
+    pub const FAILURE: Self = Self::Failure;
+
+    #[allow(dead_code, reason = "not always used due to conditional compilation")]
+    fn to_semihosting_code(self) -> i32 {
+        match self {
+            Self::Success => 0,
+            Self::Failure => 1,
+        }
+    }
+}
 
+/// Terminates the debug output session.
```

Comment:
Should it say "and leaves the MCU running" (in which case some impls may be wrong) or should it say "and may or may not reboot"?

I'd prefer the former unless there is a documented (here?) reason to do the latter.

Reply:
Pinging @bergzand to verify that the storage example on his hardware does still reboot. (Nonetheless, it may make sense to document that this does *not* terminate execution of the current function, thread or task)